src/common/mod.rs

```diff
@@ -107,40 +107,22 @@ impl CoseError {
     }
 }
 
-/// Newtype wrapper around a byte slice to allow left-over data to be detected.
-struct MeasuringReader<'a>(&'a [u8]);
-
-impl<'a> MeasuringReader<'a> {
-    fn new(buf: &'a [u8]) -> MeasuringReader<'a> {
-        MeasuringReader(buf)
-    }
-
-    fn is_empty(&self) -> bool {
-        self.0.is_empty()
-    }
-}
-
-impl<'a> ciborium_io::Read for &mut MeasuringReader<'a> {
-    type Error = EndOfFile;
-
-    fn read_exact(&mut self, data: &mut [u8]) -> Result<(), Self::Error> {
-        if data.len() > self.0.len() {
-            return Err(EndOfFile);
-        }
-
-        let (prefix, suffix) = self.0.split_at(data.len());
-        data.copy_from_slice(prefix);
-        self.0 = suffix;
-        Ok(())
+/// Convert a Ciborium error to one that uses our EndOfFile marker.
```

Comment:
super-nit: write [&#96;EndOfFile&#96;] (with square brackets and backticks) so doc comment uses a code font and is hyperlinked.

Reply:
Done, kinda.